File: app/src/main/java/com/example/jongjis/view/activity/Home/HomeActivity.java
package com.example.jongjis.view.activity.Home;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GravityCompat;
import androidx.databinding.DataBindingUtil;
import androidx.drawerlayout.widget.DrawerLayout;

import android.os.Bundle;
import android.view.View;

import com.example.jongjis.R;
import com.example.jongjis.view.adapter.HometapAdapter;
import com.example.jongjis.databinding.ActivityHomeBinding;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

public class HomeActivity extends AppCompatActivity {

    private ActivityHomeBinding binding;
    private HometapAdapter hometapAdapter;
    DrawerLayout drawerLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_home);
        drawerLayout = findViewById(R.id.drawer_layout);
        setUpTab(new HometapAdapter(this));

        binding.crdAcc.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                    openDrawer(drawerLayout);
            }
        });
    }

    public void setUpTab(HometapAdapter hometapAdapter){
        binding.vwHomeTap.setAdapter(hometapAdapter);
        TabLayoutMediator tabLayoutMediator = new TabLayoutMediator(
                binding.tabHome, binding.vwHomeTap, new TabLayoutMediator.TabConfigurationStrategy() {
            @Override
            public void onConfigureTab(@NonNull TabLayout.Tab tab, int position) {
                switch (position) {
                    case 0:
                        tab.setText("Home");
                        break;
                    case 1:
                        tab.setText("Shop");
                        break;
                }
            }
        });
        tabLayoutMediator.attach();
    }

    private static void openDrawer(DrawerLayout drawerLayout) {
        drawerLayout.openDrawer(GravityCompat.START);
    }
}

File: app/src/main/java/com/example/jongjis/view/activity/HomeActivity.kt
package com.example.jongjis.view.activity

import android.content.Context
import android.content.Intent
import android.os.Build
import android.view.View
import androidx.core.app.ActivityCompat
import com.example.jongjis.R
import com.example.jongjis.interacter.HomeInteractor
import com.example.jongjis.presenter.HomePresenter
import com.example.jongjis.view.adapter.HometapAdapter
import com.google.android.material.tabs.TabLayoutMediator
import com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy
import kotlinx.android.synthetic.main.activity_home.*
import org.json.JSONObject

class HomeActivity : BaseActivity<HomeInteractor.View,HomeInteractor.Presenter>(),
        HomeInteractor.View {
    override var mPresenter: HomeInteractor.Presenter = HomePresenter()

    override fun getLayout(): Int  = R.layout.activity_home

    override fun onHomeResponse(data: JSONObject) {
        TODO("Not yet implemented")
    }

    override fun onHomeFailed(message: String) {
        TODO("Not yet implemented")
    }

    override fun setUpHomeTap(hometapAdapter : HometapAdapter) {
        vwHomeTap.adapter = hometapAdapter
        val tabLayoutMediator = TabLayoutMediator(
                tabHome, vwHomeTap, TabConfigurationStrategy { tab, position ->
            when (position) {
                0 -> tab.text = "Home"
                1 -> tab.text = "Shop"
            }
        })
        tabLayoutMediator.attach()
    }

    override fun initView(){
        setStatusBarColor(ActivityCompat.getColor(this, android.R.color.white))
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
        }
        setUpHomeTap(HometapAdapter(this))

    }

    companion object {

        fun launch(context: Context) {
            val intent = Intent(context, HomeActivity::class.java)
            context.startActivity(intent)
        }

    }
}